pkg/frontend/adminactions/resources_list.go

```diff
@@ -159,6 +159,24 @@ func (a *azureActions) appendAzureNetworkResources(ctx context.Context, armResou
 					Resource: rt.RouteTable,
 				})
 			}
+			//Due to BYO NSGs not belonging to the managed resource group, we would like to list them as well
+			if snet.Properties.NetworkSecurityGroup.ID != nil {
```

Comment:
Instead of nesting this further, i would suggest you swap this if-statement around like so:

```go
if snet.Properties.NetworkSecurityGroup.ID == nil {
    continue // don't do anything when it's nil
}
// do the rest that's currently inside the if-statement.
r, err := azure.ParseResourceID(*snet.Properties.NetworkSecurityGroup.ID)
....
```